connector/src/main/java/org/geysermc/connector/network/translators/chat/MessageTranslator.java

```diff
@@ -75,30 +70,43 @@
      * @param locale Locale to use for translation strings
      * @return Parsed and formatted message for bedrock
      */
-    public static String convertMessage(String message, String locale) {
-        Component component = GsonComponentSerializer.gson().deserialize(message);
+    public static String convertMessage(Component message, String locale) {
+        try {
+            // Get a Locale from the given locale string
+            Locale localeCode = Locale.forLanguageTag(locale.replace('_', '-'));
+            message = RENDERER.render(message, localeCode);
 
-        // Get a Locale from the given locale string
-        Locale localeCode = Locale.forLanguageTag(locale.replace('_', '-'));
-        component = RENDERER.render(component, localeCode);
+            String legacy = LegacyComponentSerializer.legacySection().serialize(message);
 
-        String legacy = LegacyComponentSerializer.legacySection().serialize(component);
+            // Strip strikethrough and underline as they are not supported on bedrock
+            legacy = legacy.replaceAll("\u00a7[mn]", "");
 
-        // Strip strikethrough and underline as they are not supported on bedrock
-        legacy = legacy.replaceAll("\u00a7[mn]", "");
+            // Make color codes reset formatting like Java
+            // See https://minecraft.gamepedia.com/Formatting_codes#Usage
+            legacy = legacy.replaceAll("\u00a7([0-9a-f])", "\u00a7r\u00a7$1");
+            legacy = legacy.replaceAll("\u00a7r\u00a7r", "\u00a7r");
 
-        // Make color codes reset formatting like Java
-        // See https://minecraft.gamepedia.com/Formatting_codes#Usage
-        legacy = legacy.replaceAll("\u00a7([0-9a-f])", "\u00a7r\u00a7$1");
-        legacy = legacy.replaceAll("\u00a7r\u00a7r", "\u00a7r");
+            return legacy;
+        } catch (Exception | AssertionError e) {
+            GeyserConnector.getInstance().getLogger().debug(GSON_SERIALIZER.serialize(message));
```

Comment:
Can the GSON_SERIALIZER throw an exception? If yes, we should just print the message.

Reply:
No it should never happen since we have converted from json and got a component in the protocol lib